File: backend/TirthSeva.API/Services/PaymentService.cs
using TirthSeva.API.Data;
using TirthSeva.API.Models;
using Microsoft.EntityFrameworkCore;

namespace TirthSeva.API.Services
{
    public class PaymentService
    {
        private readonly ApplicationDbContext _context;
        private readonly BookingService _bookingService;

        public PaymentService(ApplicationDbContext context, BookingService bookingService)
        {
            _context = context;
            _bookingService = bookingService;
        }

        public async Task<Payment?> ProcessPaymentAsync(int bookingId, string paymentMethod, int userId)
        {
            var booking = await _context.Bookings.FindAsync(bookingId);

            if (booking == null || booking.UserId != userId)
            {
                return null;
            }

            // Check if already paid
            var existingPayment = await _context.Payments.FirstOrDefaultAsync(p => p.BookingId == bookingId);
            if (existingPayment != null)
            {
                return null;
            }

            // Mock payment processing
            var payment = new Payment
            {
                BookingId = bookingId,
                Amount = booking.TotalAmount,
                PaymentMethod = paymentMethod,
                TransactionId = $"TXN_{Guid.NewGuid().ToString().Substring(0, 12).ToUpper()}",
                Status = "Success",
                CreatedAt = DateTime.UtcNow,
                CompletedAt = DateTime.UtcNow
            };

            _context.Payments.Add(payment);
            
            // Update booking payment status
            await _bookingService.UpdatePaymentStatusAsync(bookingId, "Completed");

            await _context.SaveChangesAsync();

            return payment;
        }

        public async Task<Payment?> GetPaymentByBookingIdAsync(int bookingId)
        {
            return await _context.Payments
                .FirstOrDefaultAsync(p => p.BookingId == bookingId);
        }

        public async Task<List<Payment>> GetPaymentHistoryAsync(int userId)
        {
            return await _context.Payments
                .Include(p => p.Booking)
                .Where(p => p.Booking.UserId == userId)
                .OrderByDescending(p => p.CreatedAt)
                .ToListAsync();
        }

        public async Task<bool> VerifyPaymentAsync(string transactionId)
        {
            var payment = await _context.Payments
                .FirstOrDefaultAsync(p => p.TransactionId == transactionId);

            return payment != null && payment.Status == "Success";
        }
    }
}

File: backend-spring/src/main/java/com/tirthseva/api/service/PaymentService.java
package com.tirthseva.api.service;

import com.tirthseva.api.entity.Booking;
import com.tirthseva.api.entity.Payment;
import com.tirthseva.api.repository.BookingRepository;
import com.tirthseva.api.repository.PaymentRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class PaymentService {

    private final PaymentRepository paymentRepository;
    private final BookingRepository bookingRepository;
    private final BookingService bookingService;

    @Transactional
    public Payment processPayment(Integer bookingId, String paymentMethod, Integer userId) {
        Booking booking = bookingRepository.findById(bookingId).orElse(null);

        if (booking == null || !booking.getUserId().equals(userId)) {
            return null;
        }

        // Check if already paid
        if (paymentRepository.findByBookingId(bookingId).isPresent()) {
            return null;
        }

        // Mock payment processing
        Payment payment = Payment.builder()
                .bookingId(bookingId)
                .amount(booking.getTotalAmount())
                .paymentMethod(paymentMethod)
                .transactionId("TXN_" + UUID.randomUUID().toString().substring(0, 12).toUpperCase())
                .status("Success")
                .createdAt(LocalDateTime.now())
                .completedAt(LocalDateTime.now())
                .build();

        paymentRepository.save(payment);

        // Update booking payment status
        bookingService.updatePaymentStatus(bookingId, "Completed");

        return payment;
    }

    public Payment getPaymentByBookingId(Integer bookingId) {
        return paymentRepository.findByBookingId(bookingId).orElse(null);
    }

    public List<Payment> getPaymentHistory(Integer userId) {
        return paymentRepository.findByUserIdOrderByCreatedAtDesc(userId);
    }

    public boolean verifyPayment(String transactionId) {
        Payment payment = paymentRepository.findByTransactionId(transactionId).orElse(null);
        return payment != null && "Success".equals(payment.getStatus());
    }
}
